### PR TITLE
Servlets: additional features

### PR DESCRIPTION
**Pull request by bori00**
The user can now **set the number of comments that they want to see** and they can also **delete all the comments**

The comments page has to additional elements: 
* number input to set the number of comments that the user wants to see: whenever a change is detected, the comments are fetched again from the server, but with an additional parameter - the maximum number of comments to display(if there are less comments than specified, then all the comments will be displayed). The parameter is sent via the url, using the GET method
* a button to delete all of the comments: when the button is clicked, firstly, the DeleteCommentsServlet's doPost() method is called, and that deletes the comments from the database. Secondly, all the currently displayed comments(displayed on cards) are removed from the DOM

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+/* This servlet is responsible for deleting all the comments from the database */
+@WebServlet("/delete-comments-data")
+public class DeleteCommentsServlet extends HttpServlet {
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery comments = getCommentsFromDatastore(); 
+        for (Entity entity : comments.asIterable()) {
+           datastore.delete(entity.getKey());
```
Comment (cgollner) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java:42]: Wondering if there is no method to delete all entities of type "Comment" for example?

File: portfolio/src/main/webapp/comments.js [hunk 2]
```diff
@@ -53,4 +71,15 @@ function addAllCommentsToDOM(comments){
         addCommentToDOM(comments[i]);
         console.log(comments[i]);
     }
+}
+
+function removeAllCommentsFromDOM(){
+    var commentCardHolders = document.getElementsByClassName("card-holder-small");
+    console.log(commentCardHolders.length + " cards on the page");
+    var noCards = commentCardHolders.length;
+    for(var i=noCards-1; i>=0; i--){
```
Comment (cgollner) [inline portfolio/src/main/webapp/comments.js]: Wondering if we could simply remove all children from "main-comments" so we don't have to individually remove them? Is there such a DOM function?

File: portfolio/src/main/webapp/comments.js [hunk 3]
```diff
@@ -1,12 +1,30 @@
...
+    var maxCommentsInput = document.getElementById("max-comments");
+    maxCommentsInput.addEventListener("change", displayComments);
+}
+
 function displayComments(){
-    fetch('/comments-data')
+    removeAllCommentsFromDOM();
+    var maxComments = getNoMaxComments();
+    console.log(maxComments);
+    fetch('/comments-data?max-comments='+maxComments)
```
Comment (cgollner) [inline portfolio/src/main/webapp/comments.js]: Is there a way to nicely set the arguments without having to manually to string concatenation?

**Review by cgollner — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/comments.js (hunk 3 above)
Comment (bori00) [reply portfolio/src/main/webapp/comments.js]: I updated that part using a URL object and the searchParams.append() function

File: portfolio/src/main/webapp/comments.js (hunk 2 above)
Comment (bori00) [reply portfolio/src/main/webapp/comments.js]: I restructured a little the main element, and then I set innerHTML to an empty string to remove all the children

**Review by cgollner — APPROVED** [review]

**Pull request merged**